src/ATen/test/basic.cpp

```diff
@@ -33,11 +33,11 @@ static void test(Type & type) {
     std::cout << "ones and dot:" << std::endl;
     Tensor b = type.ones({3, 4});
     std::cout << b << std::endl;
-    ASSERT(24 == (b+b).sum().toDouble());
+    ASSERT(24 == Scalar((b+b).sum()).toDouble());
     std::cout << b.numel() << std::endl;
     ASSERT(12 == b.numel());
     std::cout << b.dot(b) << std::endl;
-    ASSERT(b.dot(b).toDouble() == 12);
+    ASSERT(Scalar(b.dot(b)).toDouble() == 12);
```

Comment:
For direct users of ATen this is pretty annoying. Can we add the helpers toDouble/toFloat, etc. directly to Tensor (in this case they should just do Scalar(fo).toX())?

Reply:
I agree this is a good idea, see my message above.  I'm a bit concerned that users would think toFloat would be defined as toType(kFloat), not Scalar(*this).toFloat().  What do you think?  We could alternatively call this something like toFloatScalar() or (if people think that would return a Scalar) toCFloat()?

Reply:
Yeah, we should be specific, maybe `toNativeFloat()`? `toFloatScalar()` does sound like it will return a `Scalar`.

Reply:
I have a slight preference for toCFloat() over toNativeFloat() because it's shorter and native already has a meaning in ATen (native functions), but I'm happy to go with it if that's your preference.

Reply:
toCFloat is good